### PR TITLE
Add a SpanExporter interface and a simple span-processor implementation.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

**Comment by bogdandrutu** [comment]
Friendly ping here.

File: sdk/src/main/java/io/opentelemetry/sdk/trace/export/SimpleSampledSpansProcessor.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+  /** Builder class for {@link SimpleSampledSpansProcessor}. */
+  public static final class Builder {
+    private final SpanExporter spanExporter;
+
+    private Builder(SpanExporter spanExporter) {
+      this.spanExporter = Utils.checkNotNull(spanExporter, "spanExporter");
+    }
+
+    // TODO: Add metrics for total exported spans.
```
Comment (pavolloffay) [inline sdk/src/main/java/io/opentelemetry/sdk/trace/export/SimpleSampledSpansProcessor.java]: Are there issues for these todos?

File: sdk/src/main/java/io/opentelemetry/sdk/trace/export/SpanExporter.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+// TODO: Change {@code BatchSampledSpansExporter} to {@link BatchSampledSpansExporter} when the
+//  class is available.
+public interface SpanExporter {
+
+  /**
+   * Called to export sampled {@code Span}s.
+   *
+   * @param spans the list of sampled Spans to be exported.
+   */
+  void export(List<Span> spans);
```
Comment (pavolloffay) [inline sdk/src/main/java/io/opentelemetry/sdk/trace/export/SpanExporter.java]: Why not a single span? Is supposed the span processor batch spans? That interface reports only a single object.

Comment (carlosalberto) [reply sdk/src/main/java/io/opentelemetry/sdk/trace/export/SpanExporter.java]: (To me it sounds like the `Simple` processor is a kind of toy to show how things work - in a more advanced scenario, a not-so-simple processor ought to buffer them instead)

Comment (bogdandrutu) [reply sdk/src/main/java/io/opentelemetry/sdk/trace/export/SpanExporter.java]: Correct, we will have a batch version that will be more complex.

Comment (bogdandrutu) [reply sdk/src/main/java/io/opentelemetry/sdk/trace/export/SpanExporter.java]: Also may be used in environments where batching may be a problem (like cloud functions, appengine).

File: sdk/src/main/java/io/opentelemetry/sdk/trace/export/SimpleSampledSpansProcessor.java (hunk 1 above)
Comment (bogdandrutu) [reply sdk/src/main/java/io/opentelemetry/sdk/trace/export/SimpleSampledSpansProcessor.java]: No, should I create?

Comment (pavolloffay) [reply sdk/src/main/java/io/opentelemetry/sdk/trace/export/SimpleSampledSpansProcessor.java]: It makes it easier to track if it's  important.

**Pull request merged**